Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -441,10 +441,13 @@ fileprivate extension Array<AnnotatedTestItem> {
         // as the root item.
         if rootItem.isExtension && !item.isExtension {
           var newItem = item
-          newItem.testItem.children += rootItem.testItem.children
+          newItem.testItem.children = (newItem.testItem.children + rootItem.testItem.children).deduplicateIds()
           rootItem = newItem
+        } else if rootItem.testItem.children.isEmpty && item.testItem.children.isEmpty {
+          itemDict[item.testItem.ambiguousTestDifferentiator] = item
+          continue
         } else {
-          rootItem.testItem.children += item.testItem.children
+          rootItem.testItem.children = (rootItem.testItem.children + item.testItem.children).deduplicateIds()
         }
```

Comment:
Shouldn’t we also need to deduplicate IDs of children if we add the test item for the first time (`itemDict[id] = item`). I currently don’t understand how we de-duplicate in the following case

```swift
struct MySuite {
  @Test(arguments: [1, 2, 3])
  func foo(_ x: Int) {}
  @Test(arguments: ["a", "b", "c"])
  func foo(_ x: String) {}
}
```

Reply:
`mergingTestsInExtensions` is called [recursively down the tree of children](https://github.com/swiftlang/sourcekit-lsp/pull/1662/files/65c7c181f855130cb1b10ede20e1e0cb2da6d8b9#diff-fdb0d2e4e192a73ab75b193c3a9cbadda5365e471d2aa135cd346c825d7f1abdR487), so the `result.deduplicateIds()` on line 490 handles this case.

Reply:
But don’t we only get into that recursive branch when `mergedIds.contains($0.testItem.id)`, which wouldn’t be the case in my example above?

Reply:
Ah, you're correct, I misunderstood, I'll add a test case